bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/RenaultHandler.java

```diff
@@ -0,0 +1,147 @@
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RenaultHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Doug Culnane - Initial contribution
+ */
+@NonNullByDefault
+public class RenaultHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(RenaultHandler.class);
+
+    private RenaultConfiguration config = new RenaultConfiguration();
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private HttpClient httpClient;
+
+    public RenaultHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // This binding only polls status data automatically.
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        this.config = getConfigAs(RenaultConfiguration.class);
+
+        // Validate configuration
+        if (this.config.myRenaultUsername.trim().length() == 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "MyRenault Username is empty!");
+            return;
+        }
+        if (this.config.myRenaultPassword.trim().length() == 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "MyRenault Password is empty!");
+            return;
+        }
+        if (this.config.locale.trim().length() == 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Location is empty!");
+            return;
+        }
+        if (this.config.vin.trim().length() == 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "VIN is empty!");
+            return;
+        }
+        if (this.config.refreshInterval < 1) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "The refresh interval mush to be larger than 1");
+            return;
+        }
+
+        // Background initialization:
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(this::getStatus, 0, config.refreshInterval, TimeUnit.MINUTES);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+        super.dispose();
+    }
+
+    private void getStatus() {
+        MyRenaultHttpSession httpSession = new MyRenaultHttpSession(this.config, httpClient);
+        try {
+            httpSession.initSesssion();
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
+            httpSession = null;
+            logger.error("Error My Renault Http Session.", e);
```

Comment:
Please do not use ERROR level, so rather WARNING.

Reply:
Done